src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs

```diff
@@ -531,9 +532,18 @@ public async Task StopRecord()
 			{
 				IsRecordingVideo = false;
 			}
+			try
+			{
+				CloseSession();
```

Comment:
`PrepareSession` will invoke `CloseSession`, is this neccesary?

Reply:
It isn't 
The only necessary thing is to raise the event with the file name before creating a new MediaRecorder which creates a new file and that name will be used for the event which resulted in a file which has 85 bytes in size as the "recorder file"
I also wonder if mediaRecorder.Prepare is necessary before the recording starts as it will actually create the file (85 bytes in size) before starting the recording when you are only in preview mode. The scenario where this will create problems is when you want to delete the files in the folder before starting to record a new one and you will endup deleting the file that media recorder has to use when you start the recording. I think Prepare should be called right before StartRecord but I haven't done any test on performance impact.